addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1095,6 +1101,66 @@ var SnippetsMenu = Widget.extend({
         return this.$editor.find(this.selectorEditableArea)
             .add(this.$editor.filter(this.selectorEditableArea));
     },
+    /**
+     * Manages overlays visibility towards animations/transitions
+     * states of snippets. This prevents having the overlay lagging behind the snippet
+     * css animation (with animation previews from 'website_animate' among others).
+     *
+     * @private
+     * @param {jQuery} $snippet
+     * @param {number} [animationsCountDelta=0] The delta to apply on the animations count as integer
+     */
+    manageOverlayVisibilityTowardsSnippetAnimations: function ($snippet, animationsCountDelta = 0) {
+        const positionalId = this._getPositionalId($snippet);
```

Comment:
In order to prevent showing overlays of child snippets (when having selected it and moving mouse to left panel on the parent animation option and moving back to child snippet option section) when parent snippet is animated, the only solution I found was to prevent all overlays from being displayed as long as one animation is running. With that in mind I found that SnippetsMenu was the solution as this is where SnippetEditors are created. This is also the right spot to store the dictionary of active animations and the right spot to show and hide $snippetEditorArea.

As for the key I indeed forgot that snippets could move... What we need is a unique identifier per snippet. Any Idea?